auth/constructor.go

```diff
@@ -0,0 +1,191 @@
+package auth
+
+import (
+	"fmt"
+
+	"github.com/xmidt-org/themis/xlog"
+	"github.com/xmidt-org/webpa-common/basculechecks"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	gokitprometheus "github.com/go-kit/kit/metrics/prometheus"
+	"github.com/justinas/alice"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/xmidt-org/bascule"
+	"github.com/xmidt-org/bascule/basculehttp"
+	"github.com/xmidt-org/bascule/key"
+	"github.com/xmidt-org/webpa-common/basculemetrics"
+	"go.uber.org/fx"
+)
+
+type PrimaryCOptionsIn struct {
+	fx.In
+	Logger  log.Logger
+	Options []basculehttp.COption `group:"primary_bascule_constructor_options"`
+}
+
+type PrimaryTokenFactoryIn struct {
+	fx.In
+	Logger       log.Logger
+	DefaultKeyID string         `name:"primary_bearer_default_kid"`
+	Resolver     key.Resolver   `name:"primary_bearer_key_resolver" optional:"true"`
+	Leeway       bascule.Leeway `name:"primary_bearer_leeway"`
+}
+
+type PrimaryBasculeMetricListenerIn struct {
+	fx.In
+	Listener *basculemetrics.MetricListener `name:"primary_bascule_metric_listener"`
+}
+
+func ProvidePrimaryBasculeConstructor() fx.Option {
+	return fx.Provide(
+		fx.Annotated{
+			Group: "primary_bascule_constructor_options",
+			Target: func(in PrimaryBasculeMetricListenerIn) basculehttp.COption {
+				return basculehttp.WithCErrorResponseFunc(in.Listener.OnErrorResponse)
+			},
+		},
```

Comment:
We're missing `basculehttp.WithParseURLFunc()`, it is part of the original auth.go and helps us with capability check bucketing / regex, I believe.

Reply:
Ok, let me check on that.